src/Sdk/DTLogging/Logging/ValueEncoders.cs

```diff
@@ -80,6 +81,65 @@ public static String TrimDoubleQuotes(String value)
             return trimmed;
         }
 
+        public static String PowerShellPreAmpersandPlus(String value)
+        {
+            // if the secret is passed to PS as a command and it causes an error, sections of it can be surrounded by color codes
+            // or printed individually. 
+            
+            // The secret secretpart1&secretpart2&secretpart3 would be split into 2 sections:
+            // 'secretpart1&secretpart2&' and 'secretpart3'. This method masks for the first section.
+            
+            // The secret secretpart1&+secretpart2&secretpart3 would be split into 2 sections:
+            // 'secretpart1&+' and (no 's') 'ecretpart2&secretpart3'. This method masks for the first section.
+
+            var trimmed = string.Empty;
+            if (!string.IsNullOrEmpty(value) && value.Contains("&"))
+            {
+                var secretSection = string.Empty;
+                if (value.Contains("&+"))
+                {
+                    secretSection = value.Substring(0, value.IndexOf("&+") + "&+".Length);
+                }
+                else
```

Comment:
```
else if (value.Contains("&") 
{

}
```
Is a little cleaner. There isn't a functional different here due to line 109, but this sets secretSection equal to the first character for strings not containing "&"